### PR TITLE
Add support for iteration control to PFASST.

### PR DESCRIPTION
**Pull request by memmett**
This is a WIP until tests are added.  This addressed #125.

### CODE REVIEW

File: include/pfasst/mpi_communicator.hpp [hunk 1]
```diff
@@ -61,11 +62,95 @@ namespace pfasst
...
+
+        virtual void post()
+        {
+          // we'll use blocking send/recv for status info
+        }
+
+        virtual void send()
+        {
+          if (mpi->size() == 1) { return; }
+          if (mpi->rank() == mpi->size() - 1) { return; }
```
Comment (torbjoernk) [inline include/pfasst/mpi_communicator.hpp]: @memmett, could you comment this logic a little bit (in source)? It doesn't seem that obvious to me.

**Comment by memmett** [comment]
@pint-jenkins retest this please

**Comment by memmett** [comment]
@torbjoernk Jenkins seems to be broken...

**Comment by memmett** [comment]
This closes #125.

**Pull request merged**